src/interrupt.rs

```diff
@@ -39,23 +39,38 @@ pub unsafe fn enable() {
 /// Execute closure `f` in an interrupt-free context.
 ///
 /// This as also known as a "critical section".
+#[deprecated(since = "0.4.0", note = "critical_section::with() allows alternate implementations; interrupt:free() is a hardcoded implementation of critical_section::with() with a different type signature.")]
 pub fn free<F, R>(f: F) -> R
 where
     F: for<'a> FnOnce(&'a CriticalSection<'a>) -> R,
 {
-    let status = ::register::sr::read();
-
     // disable interrupts
-    disable();
+    let status = unsafe { acquire() };
 
     let cs = unsafe { CriticalSection::new() };
     let r = f(&cs);
 
     // If the interrupts were active before our `disable` call, then re-enable
     // them. Otherwise, keep them disabled
```

Comment:
Delete this comment

Reply:
Why does this comment need to be deleted? I've fixed the other things locally.

Reply:
The logic of checking the status to decide whether to re-enable the interrupt is inside the `release` function, so this comment should be moved there.

Also, why isn't `acquire` used in this call?